sdk/core/Azure.Core/src/RehydrationToken.Serialization.cs

```diff
@@ -0,0 +1,121 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.ClientModel.Primitives;
+using System.Text.Json;
+
+#nullable enable
+
+namespace Azure.Core
+{
+    public partial struct RehydrationToken : IJsonModel<RehydrationToken>
+    {
+        internal RehydrationToken DeserializeRehydrationToken(JsonElement element, ModelReaderWriterOptions options)
+        {
+            if (element.ValueKind == JsonValueKind.Null)
+            {
+                return default;
```

Comment:
What happens when the JSON value is null?  Is this valid for service APIs?  What is the end-user experience if we return default from the serialization routine?

Reply:
i.e. would it make sense to throw in this case instead of returning `default`?

Reply:
You are right, we should throw on null input.